data_steward/retraction/retract_utils.py

```diff
@@ -79,6 +79,51 @@ def get_src_id(mapping_type):
     return src_id
 
 
+def get_datasets_list(project_id, dataset_ids):
+    # initialize list of all datasets in project
+    all_dataset_ids = [
+        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+    ]
+
+    if dataset_ids == 'all_datasets':
+        dataset_ids = all_dataset_ids
+    elif dataset_ids == 'none':
+        dataset_ids = []
+    else:
+        dataset_ids = dataset_ids.split()
+        # only consider datasets that exist in the project
+        dataset_ids = [
+            dataset_id for dataset_id in dataset_ids
+            if dataset_id in all_dataset_ids
+        ]
```

Comment:
I carried this over from `retract_data_bq` and there is an option for `none` to indicate no datasets, to allow retraction only from buckets. I believe this script was using empty lists earlier, but that created some confusion while running it by other devs. I wanted the intent to be explicit from the user running the tool, to avoid any ambiguity.

Reply:
hmm, this may be something to discuss at the eng sync.  I would assume that if we are retracting from a bucket only we could send a falsy value for the dataset option and get the empty list back.  Alternatively, if we really want a flag for 'none' string, I would treat it the same as the `all_datasets` flag and define it at the module level or in a constants file.

I'm assuming the program flow is to send everything through `get_datasets_list` and let the function sort out proper return values.  If this is correct, I would still keep the 'falsy' condition and add the check for 'none' to it.  With the current definition, someone could send an `[]` or `None` to the function (there are no checks against this) and line 93 would attempt to split a non-string object, causing errors.